From 317defa29c5632da3c6aac1e0c5490c1defb1327 Mon Sep 17 00:00:00 2001
From: "Lu.Jiang" <lu.jiang@windriver.com>
Date: Fri, 5 Sep 2014 18:14:52 +0800
Subject: [PATCH 287/987] Move operation mapping index to iommu.h.

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

Represent operation mappings as an enum. These have been moved to iommu.h
to support IOMMU API for setting operation mappings per window.

Signed-off-by: Varun Sethi <Varun.Sethi@freescale.com>
Signed-off-by: Lu.Jiang <lu.jiang@windriver.com>
[Fix context to apply to WRL.]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/iommu/fsl_pamu.c | 5 -----
 include/linux/iommu.h    | 8 ++++++++
 2 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/drivers/iommu/fsl_pamu.c b/drivers/iommu/fsl_pamu.c
index 4611abd..a4ffe76 100644
--- a/drivers/iommu/fsl_pamu.c
+++ b/drivers/iommu/fsl_pamu.c
@@ -26,11 +26,6 @@
 #include <asm/mpc85xx.h>
 #include <asm/fsl_guts.h>
 
-/* define indexes for each operation mapping scenario */
-#define OMI_QMAN        0x00
-#define OMI_FMAN        0x01
-#define OMI_QMAN_PRIV   0x02
-#define OMI_CAAM        0x03
 
 #define make64(high, low) (((u64)(high) << 32) | (low))
 
diff --git a/include/linux/iommu.h b/include/linux/iommu.h
index 1da1915..0c519bf 100644
--- a/include/linux/iommu.h
+++ b/include/linux/iommu.h
@@ -90,6 +90,14 @@ enum iommu_cap {
 	IOMMU_CAP_NOEXEC,		/* IOMMU_NOEXEC flag */
 };
 
+/* define indexes for each operation mapping scenario */
+enum omap_index {
+	OMI_QMAN,
+	OMI_FMAN,
+	OMI_QMAN_PRIV,
+	OMI_CAAM,
+};
+
 /*
  * Following constraints are specifc to FSL_PAMUV1:
  *  -aperture must be power of 2, and naturally aligned
-- 
1.9.1

